full_gpu_inference_pipeline/README.md

```diff
@@ -0,0 +1,177 @@
+# Deploy MONAI pipeline by Triton and run the full pipeline on GPU step by step
+- [Deploy MONAI pipeline by Triton and run the full pipeline on GPU step by step](#deploy-monai-pipeline-by-triton-and-run-the-full-pipeline-on-gpu-step-by-step)
+  * [overview](#overview)
+  * [Prepare the model repository](#prepare-the-model-repository)
+    + [Prepare the model repository file directories](#prepare-the-model-repository-file-directories)
+  * [Environment Setup](#environment-setup)
+    + [Setup Triton environment](#setup-triton-environment)
+    + [Setup python execution environment](#setup-python-execution-environment)
+  * [Benchmark](#benchmark)
+    + [Understanding the benchmark output](#understanding-the-benchmark-output)
+    + [HTTP vs. gRPC vs. shared memory](#http-vs-grpc-vs-shared-memory)
+    + [Pre/Post-processing on GPU vs. CPU](#pre-post-processing-on-gpu-vs-cpu)
+
+## overview
+
+This repository is to implement 3D medical imaging AI inference pipeline using the model and transforms of MONAI and Triton. the goal of it is to test the influence brought by different features in MONAI and Triton for medical imaging AI inference, especially for the pipelines with large 3D images.
+
+In this repository, I will try and compare the following features:
+- [Python backend BLS](https://github.com/triton-inference-server/python_backend#business-logic-scripting) (Triton)
+- Transforms on GPU and CPU (MONAI)
+
+Before start, I highly recommand you to read the the following to links to catch up the basic features as well as usage of Triton python backend and MONAI:
+- https://github.com/triton-inference-server/python_backend
+- https://github.com/Project-MONAI/tutorials/blob/master/acceleration/fast_model_training_guide.md
+
+## Prepare the model repository
+The full pipeline is as below:
+
+<img src="https://github.com/zmsunnyday/tutorials/raw/full_gpu_inference_pipeline/full_gpu_inference_pipeline/pics/Picture3.png">
+
+### Prepare the model repository file directories
+The model repository of the experiments can be fast set up by: 
+```bash
+git clone https://gitlab-master.nvidia.com/menzhang/triton_monai.git
+cd triton_monai/
+```
+The model repository is in folder triton_models. The file structure of the model repository should be:
+```
+triton_models/
```

Comment:
thanks, could you please 

- put the model folder `triton_models/*` on google drive, and share only a URL here, like this data section https://github.com/Project-MONAI/tutorials/tree/master/self_supervised_pretraining#1data
- update to have a monai copyright section at the beginning of the files? https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md#checking-the-coding-style
- there are some gitlab links which won't work properly...

Reply:
Hi, thanks for your comments. 
-  Although `triton_model` is the model folder, there are code and config files, which I think would be better put on Github.
- I have updated MONAI copyright section at the beginning of code files.
- The Gitlab links are removed and replaced with the right links.
- Code style has been checked by following https://github.com/Project-MONAI/MONAI/blob/dev/CONTRIBUTING.md#checking-the-coding-style, and passed. 
Please feel free to let me know if there is anything that should be modified.

Reply:
Thank you, it looks good. Could you please replace the URLs of your personal pages (such as https://github.com/zmsunnyday/tutorials/raw/full_gpu_inference_pipeline/full_gpu_inference_pipeline) with the project-monai links?

for the model files, if you could send me a zip or tar file, I can upload it to https://github.com/Project-MONAI/MONAI-extra-test-data/releases/tag/0.8.1 where we agreed to use it for testing/tutorial data storage. Sorry about this inconvenience but if we all include the files in this codebase, it becomes very large and not easily managable quickly.

Reply:
Hi,
- I have replaced my personal repo links with project-MONAI links in README.md.
- I have uploaded the zip file [here](https://drive.google.com/file/d/1NPlbS7jmpmRE_4ey31SdveKBBKX9CFS9/view?usp=sharing). Please let me know if you have access to the file.
Thanks.

Reply:
thanks, I uploaded the model to https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/triton_models.zip

Reply:
Updated. Please review, thanks.